includes/Modules/Analytics/Advanced_Tracking/Measurement_Code_Injector.php

```diff
@@ -0,0 +1,79 @@
+<?php
+/**
+ * Class Google\Site_Kit\Modules\Analytics\Advanced_Tracking\Measurement_Code_Injector
+ *
+ * @package   Google\Site_Kit\Modules\Analytics
+ * @copyright 2020 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://sitekit.withgoogle.com
+ */
+
+namespace Google\Site_Kit\Modules\Analytics\Advanced_Tracking;
+
+/**
+ * Injects Javascript based on the current active plugins
+ *
+ * Class Injector
+ *
+ * @since n.e.x.t.
+ * @access private
+ * @ignore
+ */
+final class Measurement_Code_Injector {
+
+	/**
+	 * Holds a list of event configurations to be injected
+	 *
+	 * @since n.e.x.t.
+	 * @var array
+	 */
+	private $event_configurations;
+
+	/**
+	 * The javascript code that is injected for event tracking.
+	 *
+	 * @since n.e.x.t.
+	 * @var string
+	 */
+	private static $inject_script = <<<INJECT_SCRIPT
+let config;
+for ( config of eventConfigurations ) {
+	const thisConfig = config;
+	document.addEventListener( config.on, function( e ) {
+		if ( e.target.matches( thisConfig.selector ) ) {
+			alert( 'Got an event called: '.concat( thisConfig.action ) );
+			//record event with gtag here
+		} else if ( e.target.matches( thisConfig.selector.concat( ' *' ) ) ) {
+			alert( 'Got an event called: '.concat( thisConfig.action ) );
+			//record event with gtag here
+		}
+	}, true );
+}
```

Comment:
We'll need to thoroughly test this code, also for cross-browser compatibility. This code will run in any environment we cannot control, so it needs to be more "old school" unfortunately. For example:
* The entire code should be wrapped in an anonymous function that is then called: `( function() { ... } )();`; this way it doesn't affect the global namespace in any way.
* It shouldn't use any more modern (e.g. ES6) JavaScript patterns, e.g. `let` and `const` should be replaced with `var`.
* The `matches` function may not always be present, so we'll need to use a [Polyfill like e.g. AMP does](https://github.com/ampproject/amphtml/blob/feb792fcdc7d328073300d083a516c18c87c1194/src/dom.js#L479).

Reply:
Resolved.